index.js

```diff
@@ -0,0 +1,183 @@
+    this._port = (options.karmaConfig && options.karmaConfig.port) || 9876;
+    this._bundlePath = options.bundlePath || '';
+
+    this._assertValidOptions(options);
+
+    this._erikPath = path.join(options.bundlePath, 'erik');
+
+    this._registerTasks();
+  }
+
+  _assertValidOptions() {
+    if (typeof this._gulp !== 'object') {
+      throw new Error('`options.gulp` is of an invalid type.');
+    }
+
+    if (typeof this._watch !== 'boolean') {
+      throw new Error('`options.watch` is of an invalid type.');
+    }
+
+    if (typeof this._taskDependencies !== 'object') {
+      throw new Error('`options.taskDependencies` is of an invalid type.');
+    }
+
+    if (typeof this._localDependencies !== 'object') {
+      throw new Error('`options.localDependencies` is of an invalid type.');
+    }
+
+    if (typeof this._remoteDependencies !== 'object') {
+      throw new Error('`options.remoteDependencies` is of an invalid type.');
+    }
+
+    if (typeof this._bundledSpecPath !== 'string') {
+      throw new Error('`options.bundledSpecPath` is of an invalid type.');
+    }
+ 
+    if (typeof this._port !== 'number') {
+      throw new Error('`options.karmaConfig.port` is of an invalid type.');
+    }
+ 
+    if (typeof this._bundlePath !== 'string') {
+      throw new Error('`options.bundlePath` is of an invalid type.');
+    }
+   }
+
+  _registerTasks() {
+    this._registerFetchRemoteDeps();
+    this._registerBundleDeps();
+    this._registerCombine();
+    this._registerRunSpec();
+    this._registerErik();
+  }
+
+  _registerFetchRemoteDeps() {
+    this._gulp.task('erik-fetch-remote-deps', () => {
+      return streamqueue({
+        objectMode: true
+      }, ...this._remoteDependencies.map((urlStr) => {
+        const url = URL.parse(urlStr);
+        const base = `${url.protocol}//${url.host}`;
+        const path = url.path;
+
+        return remoteSrc([path], {
+          base
+        });
+      }))
+        .pipe(concat('remote-deps.js'))
+        .pipe(this._gulp.dest(this._erikPath));
+    });
+  }
+
+  _registerBundleDeps() {
+    this._gulp.task('erik-bundle-deps', () => {
+      return this._gulp.src([
+        'erik/remote-deps.js',
+        ...this._localDependencies
+      ])
+        .pipe(concat('bundled-deps.js'))
+        .pipe(this._gulp.dest(this._erikPath));
+    });
+  }
+
+  _registerCombine() {
+    /**
+     * Combine the bundled dependencies and specs to ensure that the dependencies are available to
+     * the test suite.
+     */
+    this._gulp.task('erik-combine', () => {
+      return this._gulp.src([
+        'erik/bundled-deps.js',
+        this._bundledSpecPath
+      ])
+        .pipe(concat('combined.js'))
+        .pipe(this._gulp.dest(this._erikPath));
+    });
+  }
+
+  _registerRunSpec() {
+    this._gulp.task('erik-run-spec', (done) => {
+      new karmaServer.start({
+        frameworks: ['jasmine'],
+        browsers: ['PhantomJS'],
+        reporters: ['mocha'],
```

Comment:
Yeah, it's just a formatting thing. This reporter is mocha-_style_, not specific to mocha specs :D.